### ISSUE TITLE
knn.predict(dataset) of sample code causes exception

### ISSUE THREAD

**Issue by stefano-lupo** [mljs/knn#10]
Was attempting to use the basic sample code provided on the [npm page](https://www.npmjs.com/package/ml-knn) for this package:

```js
const KNN = require('ml-knn')

var dataset = [[0, 0, 0], [0, 1, 1], [1, 1, 0], [2, 2, 2], [1, 2, 2], [2, 1, 2]];
var predictions = [0, 0, 0, 1, 1, 1];
var knn = new KNN(dataset, predictions);

dataset = [[0, 0, 0]];
 
var ans = knn.predict(dataset);
```
   
The above code caused the following exception:
```
/node_modules/ml-knn/lib/index.js:91
                if (bestNodes.size() < maxNodes || ownDistance < bestNodes.peek()[1]) {
                                                                                 ^

TypeError: Cannot read property '1' of undefined
    at nearestSearch (/home/stefano/coding/CS4404/CS4404-Project-1/node_modules/ml-knn/lib/index.js:91:82)
    at nearestSearch (/home/stefano/coding/CS4404/CS4404-Project-1/node_modules/ml-knn/lib/index.js:109:13)
    at nearestSearch (/home/stefano/coding/CS4404/CS4404-Project-1/node_modules/ml-knn/lib/index.js:109:13)
    at KDTree.nearest (/home/stefano/coding/CS4404/CS4404-Project-1/node_modules/ml-knn/lib/index.js:134:13)
    at getSinglePrediction (/home/stefano/coding/CS4404/CS4404-Project-1/node_modules/ml-knn/lib/index.js:384:36)
    at KNN.predict (/home/stefano/coding/CS4404/CS4404-Project-1/node_modules/ml-knn/lib/index.js:374:38)
    at Object.<anonymous> (/home/stefano/coding/CS4404/CS4404-Project-1/test.js:9:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```

Instantiating the model seems to work okay, but any use of the `predict(dataset)` function seems to cause the problem.

**Comment by targos** [comment]
Hello. Thanks for the report! There was a bug in the calculation of `k` when it is not provided in the options.
I fixed it and released `v2.1.3`.

**Comment by stefano-lupo** [comment]
Yeah just tested there - all fixed. Thank you for the quick response!
